api/src/application/user/user-service.js

```
@@ -30,49 +31,73 @@ module.exports = (bookshelf, emailService, renderTemplate) => {
-          subject: `${appName} Invitation`,
-          html: emailBody,
-        });
+        }).then(emailBody => {
+          return emailService.send({
+            to: user.get('email'),
+            subject: `${appName} Invitation`,
+            html: emailBody,
+          }).then(data => {
+            return data;
+          }).catch(error => {
+            Boom.badImplementation('Something went wrong', error);
+          });
+        }).catch(error => {
+          Boom.badImplementation('Something went wrong', error);
+        });  
+      }).catch(error => {
+        Boom.badImplementation('Something went wrong', error);
       });
     },
 
-    resendUserInvite(userId, appName, hoursTillExpiration) {
+    resendUserInvite(userId, appName, clientId, redirect_uri, scope, hoursTillExpiration) {
       return bookshelf.model('user').where({ id: userId }).fetch().then(user => {
         if (!user) {
           return Boom.notFound();
         }
+        return clientService.findByRedirectUriAndClientId(payload.client_id, payload.redirect_uri).then(clients => {
+          if (clients.models.length === 0) {
+            return Boom.badData('The provided redirect uri is invalid for the given client id.');
+          }
+          return self.sendInvite(user, appName, clientId, redirect_uri, scope, hoursTillExpiration).then(() => user);
+        });
 
-        return self.sendInvite(user, appName, hoursTillExpiration).then(() => user);
       });
     },
 
     inviteUser(payload) {
       let createdUser;
-      return self.create(
-        payload.email,
-        uuid.v4(),
-        {
-          app_metadata: payload.app_metadata || {},
-          profile : payload.profile || {}
+      return clientService.findByRedirectUriAndClientId(payload.client_id, payload.redirect_uri).then(clients => {
+        if (clients.models.length === 0) {
+          return Boom.badData('The provided redirect uri is invalid for the given client id.');
```

Same as above, throw this error instead of return